components/raftstore/src/store/peer_storage.rs

```diff
@@ -1440,7 +1442,7 @@ where
 }
 
 // When we bootstrap the region we must call this to initialize region local state first.
-pub fn write_initial_raft_state<T: Mutable>(raft_wb: &T, region_id: u64) -> Result<()> {
+pub fn write_initial_raft_state(raft_wb: &mut WriteBatch, region_id: u64) -> Result<()> {
```

Comment:
I don't see a reason to remove the type parameter here, and this method does need to be parameterized over the engine_traits traits. Still, I don't think you need to revert these since my patches also heavily edit these methods.

Reply:
Is there some type-level problem you ran into (that I will also run into) that forced you to remove the typarams here?